packages/engine/src/worker/runner/javascript/hash_util.js

```diff
@@ -111,11 +111,10 @@
     for (var j = 0; j < children.length; ++j) {
       const child = children[j];
       const name = child.name;
-      obj[name] = _vector_to_array(struct[name]);
-      if (child.type.listSize) {
-        // fixed-size-list
-        obj[name] = _fill_vector_for_fixed_size_list(obj[name], child);
-      }
+      obj[name] = _null_workaround_for_fixed_size_list(
+        _vector_to_array(struct[name]),
+        child.type,
+      );
```

Comment:
This feels even weirder to me. 

I think we should either rename the function, or raise an exception if it's called on something that isn't a fixedSizeList, and move the check:
```javascript
if (!("listSize" in field_type)) {
```
to wherever it's called.

```suggestion
      let inner = _vector_to_array(struct[name]);
      if (!("listSize" in field_type)) {
        inner = _null_workaround_for_fixed_size_list(
          inner,
          child.type
        );
      }
      obj[name] = inner;
```

Reply:
You say you have a fix almost ready, if we're completely getting rid of this function then I wouldn't worry too much about solving it.